### DISCUSSION TITLE
Advice please - Home Assistant  behind CGNat but I have a CPanel shared host.

### DISCUSSION THREAD

**Discussion by abreingan** [llamafilm/tesla-http-proxy-addon#90] [General]
Hi all, and apologies for making this even more complex, but my home network and HA OS has Starlink as ISP, so one possibility is the Cloudflare Argo Tunnel as well described here (thanks). It seems Starlink don't usually block port 443 so that might work.
But I also have a CPanel host and domain name which already has Let's Encrypt certificates for the domain and subdomains (lelal.com).
But when I read in section 6 "Check on the Cloudflare dashboard that in the SSL/TLS Edge Certificates there's an active universal certificate for both example.com and *.example.com" I am worried that this will conflict.  Would I need to obtain another domain name just for this?

Is it possible to just host the certificate on my CPanel host and to run the proxy addon as part of the Home Assistant even though they have quite different IP addresses (the HA could be a sub-domain of lelal.com if that helped, but would not be accessible from the web).
 Lastly I guess there is the possibility of running the proxy on the shared host but I haven't seen anyone trying that. 
So advice please about which avenue to try first. I have been lurking here for a while as my old Tesla API kept working. Well that stopped today (NSW Australia) so another option is needed.
Thanks for any help

**Comment by llamafilm** [comment]
I’ve never tested this but I believe it should be possible to host the public key on a different server than the proxy. The proxy does not need to be reachable externally.

As another possibility, if Starlink gives you a valid IPv6 address then you might be able to do it that way. Also, not tested.

**Reply by abreingan** [reply]
Thank you - I will try and host the public key on the Cpanel host first as I don't think starlink issues IPv6 addresses. I will report back.

**Reply by abreingan** [reply]
Well I think hosting the certificate on a different server works - 
First I created the tesla.example.com domain on my shared host (and the other sub domains but I suspect they are not needed). The Cpanel then generates certificates for the domains automatically so SSL is available using AutoSSL.
Then I did the first start as described in your docs (without the Nginx SSL proxy item).
Then I copied the com.tesla.3p.public-key.pem from the home assistant share folder to the https://tesla.example.com/.well-known/appspecific/ folder on the shared host.
I then completed the rest of your DOCs. All went well including the "Enroll public key in your vehicle" step.

But the reason I say "I think" is that I then fell at the last hurdle... configuring the Tesla Custom integration. First I was on an older version, and thought that might be the problem, but I upgraded to v3.21.0. No "Use Fleet API Proxy" appears under configuration so I tried the "add entry". Pressing this does bring up "Use Fleet API Proxy" and I then filled in the data required being the email I use for my Tesla account and the refresh token obtained from the returned uri. 
However when I press submit I get "Account is already configured" and on trying again the box remains unticked.
I cannot find anything relevant to the problem online, so any ideas gratefully received.

**Reply by llamafilm** [reply]
The config flow for that integration has been buggy ever since I added the proxy feature. I keep hoping someone can figure out how to make it right. The workaround is to delete the integration and add it fresh.

**Reply by abreingan** [reply]
That fixed it, and everything seems to be working. 
So thank you very much llamafilm / Elliott for the very significant work you put it to this.
Would it be worth my while further documenting the process of an external certificate host - I am happy to do so but don't use GitHub much - would I do a pull request on the docs?
All the best 
Al

**Reply by llamafilm** [reply]
I think the GitHub wiki is a good place to document non-standard configurations. You can add a new page there. Thanks!
